src/components/TsmlUI.tsx

```diff
@@ -64,33 +63,42 @@ export default function TsmlUI({
   });
 
   const { settings, strings } = mergeSettings(userSettings);
+  const [searchParams, setSearhParams] = useSearchParams();
 
   // enable forward & back buttons
-  useEffect(() => {
-    const popstateListener = () => {
-      setState({ ...state, input: getQueryString(settings) });
-    };
-    window.addEventListener('popstate', popstateListener);
+  // useEffect(() => {
+  //   const popstateListener = () => {
+  //     setState({ ...state, input: getQueryString(settings) });
+  //   };
+  //   window.addEventListener('popstate', popstateListener);
 
-    // update canonical
-    let canonical = document.querySelector('link[rel="canonical"]');
-    if (!canonical) {
-      canonical = document.createElement('link');
-      canonical.setAttribute('rel', 'canonical');
-      document.getElementsByTagName('head')[0]?.appendChild(canonical);
-    }
-    canonical.setAttribute(
-      'href',
-      formatUrl(
-        state.input.meeting ? { meeting: state.input.meeting } : state.input,
-        settings
-      )
-    );
+  //   // update canonical
+  //   let canonical = document.querySelector('link[rel="canonical"]');
+  //   if (!canonical) {
+  //     canonical = document.createElement('link');
+  //     canonical.setAttribute('rel', 'canonical');
+  //     document.getElementsByTagName('head')[0]?.appendChild(canonical);
+  //   }
+  //   canonical.setAttribute(
+  //     'href',
+  //     formatUrl(
+  //       state.input.meeting ? { meeting: state.input.meeting } : state.input,
+  //       settings
+  //     )
+  //   );
```

Comment:
🔪

Reply:
Do we not need the canonical link set in the head?